### DISCUSSION TITLE
Model validation with all the fields together

### DISCUSSION THREAD

**Discussion by Mte90** [Laravel-Backpack/community-forum#1199] [Q&A (Help)] [answered]
So I have various repeatable fields and I need to do some calculation to check if some value from a field and from another based on some math are ok or not and show an alert.

I saw https://laravel.com/docs/11.x/validation#performing-additional-validation-on-form-requests and also how to access all the data but is not clear how to do it in backpack on the model saving.

**Comment by jcastroa87** [comment]
Hello @Mte90,

It sounds like you’re looking to implement custom validation for your repeatable fields, which is definitely achievable! Have you had a chance to try out the custom validation approach?

Cheers!

**Reply by Mte90** [reply]
I read about but I need to access all the fields and do stuff at the end so it will be better something that I can do at the end without specify a field because none of them is required.
Custom validation for repeatable fields it means a rule that I have to assign to all of them and access to the whole data but is not suitable.
I would like something whee I access to all the data, do my stuff and alert in case to stop saving the stuff.

**Comment by Mte90** [comment, marked as answer]
I was able to find the solution with chatgpt because on google I wasn't very lucky.

So in your StoreRequest a code like this is executed on saving of the CRUD so it is possible to access all the data and do what you need:

```
    public function withValidator($validator)
    {
        $validator->after(function ($validator) {
            $data = $this->all();
            dd($data);
            if ($data['campo1'] === $data['campo2']) {
                $validator->errors()->add('campo2', 'an error.');
            }
        });
    }
```
